core/include/kernel/mutex.h

```diff
@@ -76,6 +76,27 @@ void mutex_unlock_recursive(struct recursive_mutex *m);
 void mutex_lock_recursive(struct recursive_mutex *m);
 #endif
 
+/*
+ * struct mutex_pm_aware - Mutex usable in PM atomic sequence
+ *
+ * Some resources need a mutex protection for runtime operations but are
+ * also accessed during specific system power transition (PM power off,
+ * suspend and resume) that operate in atomic execution environment where
+ * non-secure world is not operational, for example in fastcall SMC entries
+ * of the PSCI services. In such case we cannot take a mutex and we expect
+ * the mutex is unlocked. Additionally a spinning lock is attempted to be
+ * locked to check the resource access consistency.
```

Comment:
Please mention that `mutex_pm_aware_lock()` is expected to panic if it detects contention with an atomic context, regardless if called from a threaded or atomic context, but that contention between two threaded contexts is OK.